### PR TITLE
enh: Improvements on sharing details

### PR DESCRIPTION
**Pull request by nfebe**
[fix: Only show share permission if re-sharable](https://github.com/nextcloud/server/pull/49351/commits/815f1eb5a9b54eaa081ad1bd4298c814032cbfb0)
[refactor: Use new ShareType across file_sharing](https://github.com/nextcloud/server/pull/49351/commits/ab82adb0b3e89643d756d9864e658f1d56cb81d7)

### CODE REVIEW

**Review by nfebe — COMMENTED** [review]
The refactor is bigger than I thought it would be. :)

**Review by skjnldsv — APPROVED** [review]

File: apps/files_sharing/src/components/SharingEntry.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
 <template>
 	<li class="sharing-entry">
 		<NcAvatar class="sharing-entry__avatar"
-			:is-no-user="share.type !== SHARE_TYPES.SHARE_TYPE_USER"
+			:is-no-user="share.type !== ShareType.User"
```
Comment (artonge) [inline apps/files_sharing/src/components/SharingEntry.vue:9]: Is it working? Don't we have to expose `ShareType` in `data`?Is it working? Don't we have to expose `ShareType` in `data`?

File: apps/files_sharing/src/components/SharingInput.vue [hunk 2]
```diff
@@ -169,19 +169,18 @@ export default {
 			}
 
 			const shareType = [
```
Comment (artonge) [inline apps/files_sharing/src/components/SharingInput.vue:171]: Can't we do something like that?

```suggestion
			const shareType = Object.values(ShareType)
```

File: apps/files_sharing/src/components/SharingInput.vue [hunk 3]
```diff
@@ -369,46 +368,41 @@ export default {
...
+			case ShareType.Deck:
 				return {
 					icon: 'icon-deck',
 					iconTitle: t('files_sharing', 'Deck board'),
 				}
-			case this.SHARE_TYPES.SHARE_TYPE_SCIENCEMESH:
-				return {
-					icon: 'icon-sciencemesh',
-					iconTitle: t('files_sharing', 'ScienceMesh'),
-				}
```
Comment (artonge) [inline apps/files_sharing/src/components/SharingInput.vue:411]: Removed by mistake? I think we should keep supporting that.

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 4]
```diff
@@ -1024,22 +1026,20 @@ export default {
...
 				return EmailIcon
-			case this.SHARE_TYPES.SHARE_TYPE_CIRCLE:
+			case ShareType.Team:
 				return CircleIcon
-			case this.SHARE_TYPES.SHARE_TYPE_ROOM:
-				return ShareIcon
-			case this.SHARE_TYPES.SHARE_TYPE_DECK:
+			case ShareType.Room:
 				return ShareIcon
-			case this.SHARE_TYPES.SHARE_TYPE_SCIENCEMESH:
```
Comment (artonge) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:1042]: Same, we should keep that.

File: apps/files_sharing/src/components/SharingEntry.vue (hunk 1 above)
Comment (nfebe) [reply apps/files_sharing/src/components/SharingEntry.vue:9]: It's in `sharesMixin`! Time to start looking towards refactoring this to use the composition API :)

File: apps/files_sharing/src/components/SharingInput.vue (hunk 3 above)
Comment (nfebe) [reply apps/files_sharing/src/components/SharingInput.vue:411]: No, I thought it's no longer a thing. The type does not exist in share types?

Comment (nfebe) [reply apps/files_sharing/src/components/SharingInput.vue:411]: See : https://nextcloud-libraries.github.io/nextcloud-sharing/enums/index.ShareType.html

https://github.com/nextcloud-libraries/nextcloud-sharing/blob/main/lib/index.ts

Comment (artonge) [reply apps/files_sharing/src/components/SharingInput.vue:411]: But it is definitely a thing in the backend. We should add it in the lib.

**Review by artonge — APPROVED** [review]

**Pull request merged**